webboot/webboot.go

```diff
@@ -212,13 +220,50 @@ func main() {
 			log.Fatal(err)
 		}
 	}
+
+
+
+
+	ifRE := regexp.MustCompilePOSIX(*ifname)
+
+
+	ifnames, err := netlink.LinkList()
+	if err != nil {
+		log.Fatalf("Can't get list of link names: %v", err)
+	}
+
+	var filteredIfs []netlink.Link
+	for _, iface := range ifnames {
+		if ifRE.MatchString(iface.Attrs().Name) {
+			filteredIfs = append(filteredIfs, iface)
+		}
+	}
+
+
+
+
+	packetTimeout := time.Duration(*timeout) * time.Second
+
+	ctx, cancel := context.WithTimeout(context.Background(), packetTimeout*time.Duration(1<<uint(*retry)))
+	defer cancel()
+
+	c := dhclient.Config{
+		Timeout: packetTimeout,
+		Retries: *retry,
+	}
+	if *verbose {
+		c.LogLevel = dhclient.LogSummary
+	}
+
 	//the Request function sets up a DHCP confifuration for all interfaces,
 	//such as eth0, which is an ethernet interface.
 	if *ipv4 || *ipv6 {
 		//ifname uses the regular expression ^[we].* to check for an interface starting with w or e such as
 		//wlan0/1, enx453243, or eth0/1
-		dhclient.Request(*ifName, *timeout, *retry, *verbose, *ipv4, *ipv6)
 
+		//dhclient.Request(*ifName, *timeout, *retry, *verbose, *ipv4, *ipv6)
+		dhclient.IfUp(*ifname)
```

Comment:
this is wrong. IfUp takes a name of an interface and you are passing it the regular expression. It's also not needed here, so not sure why you added it.